addons/terraform-ec/resources/ec-account-creds.cue

```diff
@@ -0,0 +1,17 @@
+import "strings"
+
+output: {
+	type: "raw"
```

Comment:
raw type is deprecated, use k8s-objects instead.

Reply:
Hi, I per this [guide](https://kubevela.io/docs/platform-engineers/addon/terraform) to gen terraform addon, but got following error.

```shell
$ vela addon enable terraform-ec EC_API_KEY=xxx
enable addon by local dir: terraform-ec 
Error: Internal error occurred: failed calling webhook "mutating.core.oam.dev.v1beta1.applications": Post "https://vela-core-webhook.vela-system.svc:443/mutating-core-oam-dev-v1beta1-applications?timeout=10s": dial tcp 10.43.211.17:443: connect: connection refused
Internal error occurred: failed calling webhook "mutating.core.oam.dev.v1beta1.applications": Post "https://vela-core-webhook.vela-system.svc:443/mutating-core-oam-dev-v1beta1-applications?timeout=10s": dial tcp 10.43.211.17:443: connect: connection refused
```

I notice this pr https://github.com/kubevela/catalog/pull/351. And when I use previous version before this patch, It works well.

My env, installed by velad:

```shell
$ vela version
CLI Version: v1.4.8
Core Version: v1.4.8
GitRevision: git-192dc89
GolangVersion: go1.17.12
```

Any idea? Thanks

Reply:
How do you setup your cluster and vela-core? please check it or reinstall it. I can enable this addon without problems. I'm using master.
```shell
╭─qiaozp@B-Q698Q05P-1858 ~/repo/catalog/addons ‹lunarwhite-addon-ec›
╰─$ vela addon enable terraform-ec EC_API_KEY=xxx
enable addon by local dir: terraform-ec
W0816 23:44:43.805274   71852 render.go:194] Application name terraform-ec will be overwritten with addon-terraform-ec. Consider removing metadata.name in template.
I0816 23:44:45.326786   71852 apply.go:113] "creating object" name="addon-secret-terraform-ec" resource="/v1, Kind=Secret"
Addon terraform-ec enabled successfully.
I0816 23:44:48.671889   71852 utils.go:156] find cluster gateway service vela-system/kubevela-cluster-gateway-service:9443
╭─qiaozp@B-Q698Q05P-1858 ~/repo/catalog/addons ‹lunarwhite-addon-ec›
╰─$ vela version
CLI Version: v1.5.2
Core Version: 0.1.0 (master)
GitRevision: git-3df1776
GolangVersion: go1.17.13
```

As recommended, you should use `k8s-objects`.

Reply:
I setup the dev-env by [VelaD](https://kubevela.net/docs/install) on macOS. Did you mean testing on this branch `lunarwhite-addon-ec`? It also works well with me.

Let me provide more info: 

* Terraform addon will be automatically generated by [gen.go](https://github.com/kubevela/catalog/blob/master/hack/addons/terraform/gen.go) with [resources](https://github.com/kubevela/catalog/tree/master/hack/addons/terraform/terraform-provider-scaffold/resources) cue files. Developers only need provide specific Terraform provider's [metadata](https://github.com/kubevela/catalog/blob/master/hack/addons/terraform/terraform-provider-scaffold/metadata.yaml).
* Per this [pr](https://github.com/kubevela/catalog/pull/351), `k8s-objects` now can be generated rather than `raw`.

Question is: 

* When I use previous version of [resources](https://github.com/kubevela/catalog/tree/master/hack/addons/terraform/terraform-provider-scaffold/resources) cue files before this patch, which means using `raw`, It works well. (It's what this pr doing now.) 
* But when using the latest version, which means using `k8s-objects`, it will throw error as this [comment](https://github.com/kubevela/catalog/pull/443#discussion_r946750522) shown.

Can you help me check this issue? Just per this [guide](https://kubevela.io/docs/platform-engineers/addon/terraform), and check if the generated addon can be enabled without `failed calling webhook` error. As for the metadata, you can replace the origin one `hack/addons/terraform/terraform-provider-scaffold/metadata.yaml` with this:

<details>
<summary>metadata.yaml</summary>

```yaml
name: terraform-{{ .Name }}
version: 1.0.0
description: Kubernetes Terraform Controller Provider for {{ .CloudName }}
url: https://github.com/oam-dev/terraform-controller

tags:
  - Terraform Controller Provider
  - IaC
  - Cloud Resources

deployTo:
  control_plane: true
  runtime_cluster: false

dependencies:
  - name: terraform

# -------------------------------------Configuration Metadata for a Terraform Addon-------------------------------------
# provider short name
shortCloudName: ec

# The Cloud name of the provider
completeCloudName: Elastic Cloud

# When enabling a Terraform provider, these properties need to set for authentication. For Tencent Cloud,
# name: Environment variable name when authenticating Terraform, like https://github.com/oam-dev/terraform-controller/blob/master/controllers/provider/credentials.go#L59
# secretKey: Secret key when storing authentication information in a Kubernetes, like https://github.com/oam-dev/terraform-controller/blob/master/controllers/provider/credentials.go#L109.
cloudProperties:
  - name: EC_API_KEY
    secretKey: ecApiKey
    description: Get EC_API_KEY per this guide https://registry.terraform.io/providers/elastic/ec/latest/docs
```
</details> 

Thanks

Reply:
It's scaffold's problem. Fixed here: https://github.com/kubevela/catalog/pull/446 Thanks!

Reply:
Great work!